### PR TITLE
Add resource-status

### PR DESCRIPTION
**Pull request by bergerx**
Add [resource-status](https://github.com/bergerx/kubectl-resource-status) plugin.

### CODE REVIEW

**Comment by bergerx** [comment]
> I think `resource-status` is ok, but not great. What do you think about `pretty-status`?
I prefer to keep it this way, since its functionality is explicitly focused around the "resources", "status" keyword is already overloaded (same actually true for "resource" :( ). Maybe `pretty-resource-status` but that would be too explicit and hard to type.

**Comment by ahmetb** [comment]
This plugin is dope!

Honestly, we could call it `status` as well. @corneliusweig objections?

I just tried it with knative service (KSVC == KService == Service) but it didn't work:

```
$ kubectl resource-status ksvc hello
error: template: no template "Service" associated with template "template"
See 'kubectl -h' for help and examples
```

I'm assuming the CRD name overlap (core/v1 Service and Knative service) are causing issues. I hope you follow up in an issue sepaarely.

**Comment by bergerx** [comment]
@ahmetb it should not fail anymore (just updated the version), but i think wont print helpful stuff yet. I think a path forward is to make templates overridable to allow extending for CRDs.

Overall what this does is not so different than what `-o go-template` does, mainly adds some template funcs mainly for coloring the output.

**Comment by bergerx** [comment]
@corneliusweig i believe your comments are addressed.

**Comment by bergerx** [comment]
from @ahmetb:
> Honestly, we could call it status as well. @corneliusweig objections?

I'd be so happy, I already did this as a workaround:

```
~/bin $ ln -s ~/.krew/bin/kubectl-resource_status kubectl-status
~/bin $ 
```

File: plugins/resource-status.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    Usage:
+      $ kubectl resource-status
+
+    For additional options:
+      $ kubectl resource-status --help
+      or https://github.com/bergerx/kubectl-resource-status/blob/v0.1.0/doc/USAGE.md
+
+  description: |
+    Show status details of a given resource. Doesn't work for all possible
+    resource types, workload resources are mainly supported.
```
[the remark this answers is not in the document]
Comment (bergerx) [reply plugins/resource-status.yaml]: Updated the version, now it should work on any resource, but updated the description here anyway. till not the best.

File: plugins/resource-status.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  platforms:
+  - selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/bergerx/kubectl-resource-status/releases/download/v0.1.2/resource-status_linux_amd64.tar.gz
+    sha256: "f441f485cf8984dfdba6dcd3b24a487d32cc8fcc714d2e8abbc16c374647e5e3"
+    files:
+    - from: "./resource-status"
+      to: "."
```
[the remark this answers is not in the document]
Comment (bergerx) [reply plugins/resource-status.yaml]: Done.

**Comment by corneliusweig** [comment]
I'd be open to call it `status` as well, because this does pretty much everything one would expect from such a command. So go for it!

As this is a very general name then, it would be awesome if you could make sure that your plugin is maintained for a while (I don't know exactly what duration would make sense here, but definitely in the order of years).

**Comment by bergerx** [comment]
Done.

I've been using this daily for more than a year as part of our internal tooling, i'll happily keep it running for future kubernetes versions.

I'm just not sure about the roadmap, since i've been using it mostly as is for a good while. Ahmet just mentioned using this for knative resources as well. Any contributions are more than welcome, but for such cases i'm not sure i can handle them myself.

**Comment by ahmetb** [comment]
@bergerx Beware that if/when kubectl implements a `status` subcommand, your plugin will no longer be invoked. For these sort of reasons, we recommend people to be more verbose and prefer longer names. But it's not that great UX, so there's always a trade-off.

**Comment by bergerx** [comment]
I think it's fine, we can always change the name in case of a conflict.

**Comment by corneliusweig** [comment]
/lgtm
/approve
Great to have this in!

**Pull request merged**